### PR TITLE
Docker Cloud Migration

### PR DESCRIPTION
**Pull request by revett**
<p align="center">

</p>

### Problem

Docker have officially released [Docker Cloud](https://cloud.docker.com), so this repo needs updating to use the new CLI etc.

### Notes

- Have changed the script to now accept parameters. The idea being that you `curl` the script down, run it, and then delete it.
- Also added quite a bit of logging to make it easier to work out what's going on.

### CODE REVIEW

**Comment by jskeates** [comment]
Looks good so far :+1:

File: README.md [hunk 1]
```diff
@@ -1,37 +1,72 @@
...
+At [Vidsy](http://vidsy.co) we wanted to use Docker Cloud, but also benefit from the controls and features of AWS.
+
+### Tutum?
+
+The script was created back before [Docker Cloud](https://cloud.docker.co) existed, and the platform was called [Tutum](https://tutum.co). If you are still using Tutum then please refer to [this release](https://github.com/vidsy/docker-cloud-startup/releases/tag/v1.0.0-tutum).
 
+### Credentials from S3
 
-### Implementation
+Make sure `aws-cli` is installed:
```
Comment (jskeates) [inline README.md]: Worth mentioning about needing an IAM role (or similar)?

File: script.sh [hunk 2]
```diff
@@ -1,42 +1,145 @@
...
+_output "Installing dependencies..."
 
 case "$OS_KIND" in
   debian)
-    locale-gen en_GB.UTF-8
-    apt-get update
-    apt-get install -y python-pip jq
+    locale-gen en_GB.UTF-8 > /dev/null
+    apt-get update > /dev/null
+    apt-get install -y python-pip jq curl > /dev/null
```
Comment (jskeates) [inline script.sh]: Redirecting to `/dev/null` could turn out to be annoying when debugging. For example, you'd know `apt-get install` failed, but don't have the output to help work out why.

File: script.sh [hunk 3]
```diff
@@ -1,42 +1,145 @@
...
+    yum install -y python curl
+
+    # pip & jq aren't in the yum repos
     curl -O https://bootstrap.pypa.io/get-pip.py
     python get-pip.py
     curl -Lo /usr/bin/jq https://github.com/stedolan/jq/releases/download/jq-1.5/jq-linux64
     chmod ugo+x /usr/bin/jq
-    #We also need to configure sudo for the Tutum install to work
+
+    # We also need to configure sudo for the Tutum install to work
```
Comment (jskeates) [inline script.sh]: Left over Tutum in comment

File: README.md (hunk 1 above)
Comment (revett) [inline README.md]: Yep :+1: will do this

File: script.sh (hunk 2 above)
Comment (revett) [inline script.sh]: Thought this would get rid of `stdout` but still output `stderr`?

File: script.sh (hunk 3 above)
Comment (revett) [inline script.sh]: :+1:

File: script.sh [hunk 4]
```diff
@@ -99,6 +241,14 @@ case "$OS_KIND" in
     ;;
```
Comment (jskeates) [inline script.sh]: Haven't removed `curl` (if we want to)

File: script.sh [hunk 5]
```diff
@@ -46,52 +149,91 @@ case "$OS_KIND" in
...
 
-unset AWS_DEFAULT_REGION TUTUM_USER TUTUM_APIKEY TUTUM_UUID
-pip uninstall tutum awscli -y
+_output "Cleanup instance..."
+unset AWS_DEFAULT_REGION DOCKERCLOUD_USER DOCKERCLOUD_APIKEY NODE_UUID
+pip uninstall docker-cloud awscli -y
 case "$OS_KIND" in
   debian)
-    apt-get purge -y python-pip jq
+    apt-get purge -y python-pip jq curl
```
Comment (jskeates) [inline script.sh]: imho we should leave `curl` on the system. It's going to come preinstalled most (all?) of the time, removing it when we might not even have installed it ourselves seems like a bad move..

File: script.sh (hunk 2 above)
Comment (jskeates) [inline script.sh]: That's true, but you wouldn't have any of the normal output to tell you where you got to or what happened up to that point. I guess its down to opinion, I prefer my logs to be verbose 😜 I'm easy either way

File: script.sh (hunk 5 above)
Comment (revett) [inline script.sh]: :+1:

File: script.sh (hunk 4 above)
Comment (revett) [inline script.sh]: No longer removing `curl`

**Comment by revett** [comment]
:cake:

**Pull request closed without merging**